packaging/os/apt.py

```diff
@@ -596,39 +624,27 @@ def main():
             # reopen cache w/ modified config
             cache.open(progress=None)
 
-        if p['update_cache']:
-            # Default is: always update the cache
-            cache_valid = False
+        mtimestamp, updated_cache_time = get_updated_cache_time()
+        # Cache valid time is default 0, which will update the cache if
+        #  needed and `update_cache` was set to true
+        updated_cache = False
+        if p['update_cache'] or p['cache_valid_time']:
```

Comment:
@bcoca This does change the behavior by allowing the user to simply set a `cache_valid_time` which would cause the update to happen if the time was older than what was specificed. If you'd think it best I can remove the ``or`` which would set it back to the old behavior.